### PR TITLE
Uses @duffel/api types

### PR DESCRIPTION
**Pull request by igorp1**
__What__

This PR installs the latest version of `@duffel/api` and uses its types Instead of copying the types over like we were doing on https://github.com/duffelhq/duffel-components/pull/168.

### CODE REVIEW

**Comment by igorp1** [comment]
Moved the examples outside of `src`. It doesn't really belong there and makes easier for us to ignore with our tools Screenshot 2023-11-15 at 2 43 45 PM

File: src/tests/lib/hasServiceOfSameMetadataTypeAlreadyBeenSelected.test.ts [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
 
-const availableService: OfferAvailableBaggageService = {
+const availableService: OfferAvailableServiceBaggage = {
   id: "available_service_id",
   metadata: { type: "checked" },
   // using as any because we don't need to test the whole type
 } as any;
 
-const selectedService: CreateOrderPayloadService = {
+// TODO(idp): remove this when we merge https://github.com/duffelhq/duffel-api-javascript/pull/843
```
Comment (igorp1) [inline src/tests/lib/hasServiceOfSameMetadataTypeAlreadyBeenSelected.test.ts:11]: There's a few of these around, but should be quick to update once we merge [that PR](https://github.com/duffelhq/duffel-api-javascript/pull/843).

File: src/types/DuffelAncillariesProps.ts [hunk 2]
```diff
@@ -90,19 +91,64 @@ export interface CustomStyles {
...
+
+interface CancelForAnyReasonerviceInformation
+  extends OfferAvailableServiceCFARMetadata {
+  segmentId?: undefined;
+  sliceId?: undefined;
+  total_amount: string;
+  designator?: undefined;
+  total_currency: string;
+  passengerName?: undefined;
+}
```
Comment (igorp1) [inline src/types/DuffelAncillariesProps.ts:152]: Service information is an internal concept to duffel components. So this creates an utility type to make it easier to work with it alongside the service type from @duffel/api

**Review by andrejak — APPROVED** [review]

**Comment by andrejak** [comment]
Thank you for doing this!

**Pull request merged**